ESPHome 2026.1 breaks compilation

```
src/main.cpp:44:20: error: 'font_large' was declared 'extern' and later 'static' [-fpermissive]
   44 | static font::Font *font_large;
      |                    ^~~~~~~~~~
In file included from src/esphome/components/wmbus_gateway/screen_splash.h:4,
                 from src/esphome/components/wmbus_gateway/display_manager.h:10,
                 from src/esphome.h:120,
                 from src/main.cpp:3:
src/esphome/components/wmbus_gateway/resources.h:9:29: note: previous declaration of 'font_large'
    9 | extern esphome::font::Font *font_large;
      |                             ^~~~~~~~~~
src/main.cpp:45:20: error: 'font_small' was declared 'extern' and later 'static' [-fpermissive]
   45 | static font::Font *font_small;
      |                    ^~~~~~~~~~
src/esphome/components/wmbus_gateway/resources.h:8:29: note: previous declaration of 'font_small'
    8 | extern esphome::font::Font *font_small;
      |                             ^~~~~~~~~~
src/main.cpp:46:22: error: 'qr_code' was declared 'extern' and later 'static' [-fpermissive]
   46 | static image::Image *qr_code;
      |                      ^~~~~~~
src/esphome/components/wmbus_gateway/resources.h:7:31: note: previous declaration of 'qr_code'
    7 | extern esphome::image::Image *qr_code;
      |                               ^~~~~~~
src/main.cpp:47:22: error: 'logo' was declared 'extern' and later 'static' [-fpermissive]
   47 | static image::Image *logo;
      |                      ^~~~
src/esphome/components/wmbus_gateway/resources.h:6:31: note: previous declaration of 'logo'
    6 | extern esphome::image::Image *logo;
```

--- a/components/wmbus_gateway/__init__.py
+++ b/components/wmbus_gateway/__init__.py
@@ -1,15 +1,25 @@
 import esphome.codegen as cg
+from esphome.cpp_generator import AssignmentExpression
 import esphome.config_validation as cv
-from esphome.core import CORE
+from esphome.core import CORE, ID
 from esphome.components.binary_sensor import BinarySensor
 from esphome.components.ssd1306_base import SSD1306
+from esphome.components.image import Image_
+from esphome.components.font import Font
 from esphome.components.wmbus_meter.base_sensor import BaseSensor
 from esphome.const import CONF_DISPLAY_ID, CONF_ID, CONF_PAGES
 
 gui_ns = cg.esphome_ns.namespace("wmbus_gateway")
+resources_ns = gui_ns.namespace("resources")
 DisplayManager = gui_ns.class_("DisplayManager", cg.PollingComponent)
 
 CONF_BUTTON_ID = "button_id"
+
+CONF_RESOURCES = "resources"
+CONF_LOGO = "logo"
+CONF_QRCODE = "qrcode"
+CONF_FONT_SMALL = "font_small"
+CONF_FONT_LARGE = "font_large"
 
 CODEOWNERS = ["@kubasaw"]
 AUTO_LOAD = ["network"]
@@ -20,6 +30,14 @@ CONFIG_SCHEMA = cv.Schema(
         cv.GenerateID(CONF_ID): cv.declare_id(DisplayManager),
         cv.GenerateID(CONF_DISPLAY_ID): cv.use_id(SSD1306),
         cv.GenerateID(CONF_BUTTON_ID): cv.use_id(BinarySensor),
+        cv.Required(CONF_RESOURCES): cv.Schema(
+            {
+                cv.Required(CONF_LOGO): cv.use_id(Image_),
+                cv.Required(CONF_QRCODE): cv.use_id(Image_),
+                cv.Required(CONF_FONT_SMALL): cv.use_id(Font),
+                cv.Required(CONF_FONT_LARGE): cv.use_id(Font),
+            }
+        ),
         cv.Optional(CONF_PAGES, default="all"): cv.Any(
             "all",
             [cv.use_id(BaseSensor)],
@@ -29,6 +47,16 @@ CONFIG_SCHEMA = cv.Schema(
 
 
 async def to_code(config):
+    for var_name in (CONF_LOGO, CONF_QRCODE, CONF_FONT_SMALL, CONF_FONT_LARGE):
+        cg.add(
+            AssignmentExpression(
+                None,
+                "",
+                getattr(resources_ns, var_name),
+                await cg.get_variable(config[CONF_RESOURCES][var_name]),
+            )
+        )
+
     manager = cg.new_Pvariable(
         config[CONF_ID],
         await cg.get_variable(config[CONF_DISPLAY_ID]),

--- a/components/wmbus_gateway/__init__.py
+++ b/components/wmbus_gateway/__init__.py
@@ -1,7 +1,7 @@
 import esphome.codegen as cg
 from esphome.cpp_generator import AssignmentExpression
 import esphome.config_validation as cv
-from esphome.core import CORE, ID
+from esphome.core import CORE
 from esphome.components.binary_sensor import BinarySensor
 from esphome.components.ssd1306_base import SSD1306
 from esphome.components.image import Image_

--- a/components/wmbus_gateway/resources.cpp
+++ b/components/wmbus_gateway/resources.cpp
@@ -1,0 +1,15 @@
+#include "resources.h"
+
+namespace esphome
+{
+    namespace wmbus_gateway
+    {
+        namespace resources
+        {
+            esphome::image::Image *logo = nullptr;
+            esphome::image::Image *qrcode = nullptr;
+            esphome::font::Font *font_small = nullptr;
+            esphome::font::Font *font_large = nullptr;
+        }
+    }
+}

--- a/components/wmbus_gateway/resources.h
+++ b/components/wmbus_gateway/resources.h
@@ -3,7 +3,16 @@
 #include "esphome/components/image/image.h"
 #include "esphome/components/font/font.h"
 
-extern esphome::image::Image *logo;
-extern esphome::image::Image *qr_code;
-extern esphome::font::Font *font_small;
-extern esphome::font::Font *font_large;
+namespace esphome
+{
+    namespace wmbus_gateway
+    {
+        namespace resources
+        {
+            extern esphome::image::Image *logo;
+            extern esphome::image::Image *qrcode;
+            extern esphome::font::Font *font_small;
+            extern esphome::font::Font *font_large;
+        }
+    }
+}

--- a/components/wmbus_gateway/screen_info.h
+++ b/components/wmbus_gateway/screen_info.h
@@ -21,13 +21,13 @@ namespace esphome
 
                 for (const auto &info : infos)
                 {
-                    it.print(0, y, font_small, info.first.c_str());
-                    font_small->measure((info.first + ' ').c_str(), &occupied, &dummy, &dummy, &dummy);
+                    it.print(0, y, resources::font_small, info.first.c_str());
+                    resources::font_small->measure((info.first + ' ').c_str(), &occupied, &dummy, &dummy, &dummy);
                     this->render_string(it,
                                         it.get_width(),
                                         y,
                                         display::TextAlign::TOP_RIGHT,
-                                        font_small,
+                                        resources::font_small,
                                         info.second.c_str(),
                                         it.get_width() - occupied);
                     y += it.get_height() / infos.size();

--- a/components/wmbus_gateway/screen_qrcode.h
+++ b/components/wmbus_gateway/screen_qrcode.h
@@ -12,17 +12,17 @@ namespace esphome
         protected:
             void render_fcn(display::Display &it) override
             {
-                it.print(0, 0, font_small, "More");
-                it.print(0, font_small->get_height(), font_small, "info");
-                it.print(0, font_small->get_height() * 2, font_small, "at:");
+                it.print(0, 0, resources::font_small, "More");
+                it.print(0, resources::font_small->get_height(), resources::font_small, "info");
+                it.print(0, resources::font_small->get_height() * 2, resources::font_small, "at:");
 
                 const uint8_t scale = 3;
                 const uint8_t x_offset = 65;
                 const uint8_t y_offset = 1;
 
-                for (uint8_t x = 0; x < qr_code->get_width(); x++)
-                    for (uint8_t y = 0; y < qr_code->get_height(); y++)
-                        if (qr_code->get_pixel(x, y).is_on())
+                for (uint8_t x = 0; x < resources::qrcode->get_width(); x++)
+                    for (uint8_t y = 0; y < resources::qrcode->get_height(); y++)
+                        if (resources::qrcode->get_pixel(x, y).is_on())
                             it.filled_rectangle(x * scale + x_offset, y * scale + y_offset, scale, scale);
             }
         };

--- a/components/wmbus_gateway/screen_sensor.h
+++ b/components/wmbus_gateway/screen_sensor.h
@@ -23,7 +23,7 @@ namespace esphome
         protected:
             void render_fcn(display::Display &it) override
             {
-                this->render_string(it, 0, 0, display::TextAlign::TOP_LEFT, font_small, this->sensor->get_name().c_str());
+                this->render_string(it, 0, 0, display::TextAlign::TOP_LEFT, resources::font_small, this->sensor->get_name().c_str());
                 this->render_value(it);
                 this->render_time_ago(it);
             }
@@ -49,7 +49,7 @@ namespace esphome
                 }
                 it.print(it.get_width(),
                          it.get_height() / 2,
-                         font_large,
+                         resources::font_large,
                          display::TextAlign::CENTER_RIGHT,
                          data.c_str());
             }
@@ -61,7 +61,7 @@ namespace esphome
 
                 it.print(it.get_width(),
                          it.get_height(),
-                         font_small,
+                         resources::font_small,
                          display::TextAlign::BOTTOM_RIGHT,
                          data.c_str());
             }

--- a/components/wmbus_gateway/screen_splash.h
+++ b/components/wmbus_gateway/screen_splash.h
@@ -14,9 +14,9 @@ namespace esphome
         protected:
             void render_fcn(display::Display &it) override
             {
-                it.print(it.get_width() / 2, 0, font_small, display::TextAlign::TOP_CENTER, std::strchr(ESPHOME_PROJECT_NAME, '.') + 1);
-                it.image(it.get_width() / 2, it.get_height() / 2, logo, display::ImageAlign::CENTER);
-                it.print(it.get_width() / 2, it.get_height(), font_small, display::TextAlign::BOTTOM_CENTER, ESPHOME_PROJECT_VERSION);
+                it.print(it.get_width() / 2, 0, resources::font_small, display::TextAlign::TOP_CENTER, std::strchr(ESPHOME_PROJECT_NAME, '.') + 1);
+                it.image(it.get_width() / 2, it.get_height() / 2, resources::logo, display::ImageAlign::CENTER);
+                it.print(it.get_width() / 2, it.get_height(), resources::font_small, display::TextAlign::BOTTOM_CENTER, ESPHOME_PROJECT_VERSION);
             }
         };
     }

--- a/packages/resources.yaml
+++ b/packages/resources.yaml
@@ -14,7 +14,7 @@ font:
       - GF_Latin_Core
       
 image:
-  - id: qr_code
+  - id: qrcode
     file: https://github.com/IoTLabs-pl/wM-Bus-Gateway/blob/${wmbus_gateway_version}/resources/qrcode.png?raw=true
     type: binary
     invert_alpha: True

--- a/packages/wmbus_gateway.yaml
+++ b/packages/wmbus_gateway.yaml
@@ -18,9 +18,7 @@ external_components:
   - source: 
       type: git
       url: https://github.com/IoTLabs-pl/wM-Bus-Gateway
-      # TODO: Revert before merge to master branch
-      # ref: ${wmbus_gateway_version}
-      ref: esphome_2026_01
+      ref: ${wmbus_gateway_version}
     components:
       - wmbus_gateway
 

--- a/packages/wmbus_gateway.yaml
+++ b/packages/wmbus_gateway.yaml
@@ -18,7 +18,9 @@ external_components:
   - source: 
       type: git
       url: https://github.com/IoTLabs-pl/wM-Bus-Gateway
-      ref: ${wmbus_gateway_version}
+      # TODO: Revert before merge to master branch
+      # ref: ${wmbus_gateway_version}
+      ref: esphome_2026_01
     components:
       - wmbus_gateway
 

--- a/packages/wmbus_gateway.yaml
+++ b/packages/wmbus_gateway.yaml
@@ -46,3 +46,8 @@ wmbus_radio:
               - delay: 50ms
 
 wmbus_gateway:
+  resources:
+    font_large: font_large
+    font_small: font_small
+    qrcode: qrcode
+    logo: logo

